cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md

```diff
@@ -908,6 +908,10 @@ functions that are specific to columns of Strings. These functions reside in the
 namespace. Similarly, functionality used exclusively for unit testing is in the `cudf::test::`
 namespace.
 
+The public function is expected to contain a call to `CUDF_FUNC_RANGE()` followed by a call to
+a `detail` function with same name and parameters as the public function.
+See the [Streams](#streams) section for an example of this pattern.
```

Comment:
*nit*: The example in that section shows a pattern where the public function doesn't have a stream argument, but the detail function does. So by a naive reading the example contradicts the requirement in the first sentence.